docs/src/js/base/switch.js

```diff
@@ -0,0 +1,77 @@
+/**
+* Copyright 2019 eBay Inc.
+*
+* Use of this source code is governed by a MIT-style
+* license that can be found in the LICENSE file or at
+* https://opensource.org/licenses/MIT.
+*/
+
+function onKeyDown(e) {
+    if (e.keyCode === 32) {
+        e.preventDefault();
+        this.toggle();
+    }
+}
+
+function onClick(e) {
+    this.toggle();
+}
+
+module.exports = class {
+    constructor(widgetEl) {
+        this.el = widgetEl;
+        this.controlEl = widgetEl.querySelector('span.switch__control');
+
+        this.controlEl.tabIndex = (this.disabled === true) ? '-1' : '0';
+
+        this._onKeyDownListener = onKeyDown.bind(this);
+        this._onClickListener = onClick.bind(this);
```

Comment:
What's the benefit of setting up handlers this way instead of (1) removing the non-exported functions at top, (2) removing bind code, and (3) creating the functions inside the class
```
_onKeyDownListener(e) {
  // implementation
}
_onClickListener(e) {
 // implementation
}
```

Reply:
Nevermind. I found an article that refreshed my memory on this topic. I think binding `this` is required.
https://www.freecodecamp.org/news/this-is-why-we-need-to-bind-event-handlers-in-class-components-in-react-f7ea1a6f93eb/

Reply:
This pattern is a throwback to "older days" (circa 2013) when it was an effective way to combat the problem of closures and garbage collection - particularly with event handlers. To be completely honest, I'm not sure it is necessary any more, but for the sake of consistency (all the MIND pattern example widgets are written in this way) I'm just keeping it like this for now.

I will perhaps bring them all up to date at some point, but also bear in mind that this is just quick demo code, and not intended to be "production ready". When I get to creating the  ebayui-web-component version of the switch, the code will need to be production ready.